api/environments/dynamodb/services.py

```diff
@@ -26,39 +45,55 @@ def migrate_environments_to_v2(project_id: int) -> IdentityOverridesV2Changeset
 
     logger.info("Migrating environments to v2 for project %d", project_id)
 
-    environments_to_migrate = Environment.objects.filter(project_id=project_id)
-    identity_overrides_to_migrate = _iter_paginated_overrides(
-        identity_wrapper=identity_wrapper,
-        environments=environments_to_migrate,
+    environments_to_migrate = Environment.objects.filter_for_document_builder(
+        project_id=project_id
     )
 
-    changeset = IdentityOverridesV2Changeset(
-        to_put=list(identity_overrides_to_migrate), to_delete=[]
-    )
-    logger.info(
-        "Retrieved %d identity overrides to migrate for project %d",
-        len(changeset.to_put),
-        project_id,
-    )
+    identity_overrides_changeset = IdentityOverridesV2Changeset(to_put=[], to_delete=[])
+    result_status = EdgeV2MigrationStatus.COMPLETE
+
+    try:
+        to_put = list(
+            _iter_paginated_overrides(
```

Comment:
We should use project_expression to   only fetch the data we need for migration

Reply:
Added.